waspc/data/Generator/templates/sdk/wasp/client/operations/hooks.ts

```diff
@@ -28,8 +28,12 @@ export function useQuery<Input, Output>(
   return rqUseQuery({
     // todo: The full queryCacheKey is constructed in two places, both here and
     // inside the Query. See https://github.com/wasp-lang/wasp/issues/2017
-    queryKey: makeQueryCacheKey(query, queryFnArgs),
-    queryFn: () => query(queryFnArgs),
+    // FIXME: query fns don't handle the `undefined` case correctly
+    // https://github.com/wasp-lang/wasp/issues/2017
+    queryKey: makeQueryCacheKey(query, (queryFnArgs as Input)),
+    // FIXME: query fns don't handle the `undefined` case correctly
+    // https://github.com/wasp-lang/wasp/issues/2017
+    queryFn: () => query(queryFnArgs as Input),
```

Comment:
Added the details into the issue: https://github.com/wasp-lang/wasp/issues/2017#issue-2283681572